index.ts

```diff
@@ -221,6 +224,36 @@ export class BalancerPoolsAPI extends Stack {
       }
     );
 
+    const tenderlySimulateLambda = new NodejsFunction(
+      this,
+      'tenderlySimulateFunction',
+      {
+        entry: join(__dirname, 'src', 'lambdas', 'tenderly-simulate.ts'),
+        environment: {
+          TENDERLY_USER: TENDERLY_USER || '',
+          TENDERLY_PROJECT: TENDERLY_PROJECT || '',
+          TENDERLY_ACCESS_KEY: TENDERLY_ACCESS_KEY || '',
+        },
+        runtime: Runtime.NODEJS_14_X,
+        timeout: Duration.seconds(15),
+      }
+    );
+
+    const tenderlyEncodeStatesLambda = new NodejsFunction(
+      this,
+      'tenderlyEncodeStatesFunction',
+      {
+        entry: join(__dirname, 'src', 'lambdas', 'tenderly-encode-states.ts'),
+        environment: {
+          TENDERLY_USER: TENDERLY_USER || '',
+          TENDERLY_PROJECT: TENDERLY_PROJECT || '',
+          TENDERLY_ACCESS_KEY: TENDERLY_ACCESS_KEY || '',
+        },
+        runtime: Runtime.NODEJS_14_X,
+        timeout: Duration.seconds(15),
+      }
+    );
```

Comment:
@timjrobinson could we add some rate limiting to this endpoint just to prevent ddos. Will still need to allow a relatively high throughput since this is triggered when people change numbers.

Reply:
Done. Made the endpoint /tenderly/simulate and /tenderly/contracts/encode-states now so they can be added to the same rate limit. It's set to 1000 per 5 min and it is a separate rate limit from the wallet check (so you can be blocked from tenderly but not from wallet check).

Tested with my dev account and the rate limiting worked.